examples/freertos_linux/Linux_gcc_gcp_iot/main.c

```diff
@@ -71,34 +71,38 @@ char ec_private_key_pem[PRIVATE_KEY_BUFFER_SIZE] = {0};
 void task_function_gcpiot_embedded_c(void *parameters) {
   const size_t task_delay = (size_t)parameters;
 
-  printf("Starting GCP IoT Embedded C Client...\n");
+  printf("[ INFO ] Starting GCP IoT Embedded C Client...\n");
 
+  /* 1. Initialize the IoTC library. */
   iotc_initialize();
 
+  /* 2. Create a context handle. */
   iotc_context_handle_t context_handle = iotc_create_context();
 
-  /* generate the client authentication JWT, which will serve as the MQTT
-   * password */
+  /* 3. Create a JWT to connect to GCP IoT Core MQTT Bridge. */
   char jwt[IOTC_JWT_SIZE] = {0};
   size_t bytes_written = 0;
   iotc_state_t state = iotc_create_iotcore_jwt(
-      iotc_project_id,
+      iotc_core_parameters.project_id,
       /*jwt_expiration_period_sec=*/3600, &iotc_connect_private_key_data, jwt,
       IOTC_JWT_SIZE, &bytes_written);
 
   if (IOTC_STATE_OK != state) {
-    printf("iotc_create_iotcore_jwt returned with error: %ul", state);
+    printf("[ FAIL ] iotc_create_iotcore_jwt returned with error: %ul", state);
     iotc_shutdown();
     return;
   }
 
   const uint16_t connection_timeout = 10;
   const uint16_t keepalive_timeout = 3;
 
-  iotc_connect(context_handle, /*username=*/NULL, /*password=*/jwt,
-               /*client_id=*/iotc_device_path, connection_timeout,
+  /* 4. Connect to GCP IoT Core MQTT Bridge. */
+  iotc_connect(context_handle, /*username=*/"unused", /*password=*/jwt,
```

Comment:
This needs to be pulled in, this bug caused our project a lot of grief figuring out that the NULL parameter for the username was tanking our connection routine @atigyi